packages/discovery-react-components/src/components/DocumentPreview/components/PdfHighlight/PdfHighlight.tsx

```diff
@@ -176,8 +203,15 @@ const Highlight: FC<{
               shape.className,
               active && `${base}__item--active`,
               active && activeClassName,
-              shape.facetId && `${baseHighlightColor}-${shape.facetId} highlight`,
-              shape.facetId && active && baseHighlightColorActive
+              shape.facetId && !hasOverlap && `${baseHighlightColor}-${shape.facetId} highlight`,
+              shape.facetId && !hasOverlap && active && baseHighlightColorActive,
+              shape.facetId && hasOverlap && baseOverlapHighlight,
+              shape.facetId && hasOverlap && active && baseHighlightColorActive,
+              // shape.facetId && hasOverlap && active && basePassThroughActive,
+              shape.facetId &&
+                hasOverlap &&
+                active &&
+                `${baseHighlightColor}-${shape.facetId} highlight`
```

Comment:
Ok, this has gotten way too confusing, all in the name of using `&&` everywhere. Instead, break this out into local variables and use simple `if/else`, so it's easier to read and understand.

Reply:
refactored